burn-tensor/src/tensor/base.rs

```diff
@@ -724,3 +725,22 @@ impl<const D: usize, B: ADBackend> Tensor<B, D> {
         Self::new(B::from_inner(inner.value))
     }
 }
+
+impl<const D: usize, B> Display for Tensor<B, D>
+where
+    B: Backend,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let dims = self.shape().dims;
+
+        f.write_str("Tensor {\n")?;
+        f.write_fmt(format_args!("\tshape: {dims:?},\n"))?;
+        f.write_fmt(format_args!("\tdevice: {:?},\n", self.device()))?;
+        f.write_str("}\n")?;
+
+        // TODO: figure out how to infer backend name + dtype
```

Comment:
`Backend::name() -> String` exists, so it would be pretty easy :)
Maybe `Backend::Elem` should implement display and should be used for dtype.